sklearn/compose/_column_transformer.py

```diff
@@ -453,7 +466,16 @@ def transform(self, X):
         """
         check_is_fitted(self, 'transformers_')
 
-        Xs = self._fit_transform(X, None, _transform_one, fitted=True)
+        try:
+            Xs = Parallel(n_jobs=self.n_jobs)(
+                delayed(_transform_one)(trans, X_sel, None, weight)
+                for _, trans, X_sel, weight in self._iter(
+                    X=X, fitted=True, replace_strings=True))
+        except ValueError as e:
```

Comment:
This is not covered by tests

Reply:
A test was added for `transform`, `fit_transform` and `fit` to check that they raise `ValueError(_ERR_MSG_1DCOLUMN)` correctly.